### PR TITLE
trac: init at 1.4

### PR DESCRIPTION
**Pull request by mmahut**
###### Motivation for this change

trac: init at 1.4

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/trac/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+, jinja2
+, pysqlite
+, psycopg2
+, pymysql
+, git
+, setuptools
+}:
+
+
+buildPythonPackage rec {
```
Comment (jonringer) [inline pkgs/tools/misc/trac/default.nix]:
```suggestion
buildPythonApplication rec {
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -2881,6 +2881,8 @@ in
   tsm-client = callPackage ../tools/backup/tsm-client { jdk8 = null; };
   tsm-client-withGui = callPackage ../tools/backup/tsm-client { };
 
+  trac = pythonPackages.callPackage ../tools/misc/trac { };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix]: would be nice nice for upstream to have a python3 supported installation....
```suggestion
  trac = pythonPackages.callPackage ../tools/misc/trac { };
```

Comment (mmahut) [reply pkgs/top-level/all-packages.nix]: Yeah, sadly not supported.

File: pkgs/tools/misc/trac/default.nix (hunk 1 above)
Comment (mmahut) [reply pkgs/tools/misc/trac/default.nix]: Good catch.

File: pkgs/tools/misc/trac/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    psycopg2
+    pymysql
+    git
+    setuptools
+  ];
+
+  meta = with lib; {
+    description = "Integrated SCM, wiki, issue tracker and project environment";
+    homepage = "https://trac.edgewall.org/";
+    license = licenses.bsd2;
```
Comment (vyp) [inline pkgs/tools/misc/trac/default.nix]:
```suggestion
    license = licenses.bsd3;
```

https://trac.edgewall.org/browser/tags/trac-1.4/COPYING

**Review by vyp — CHANGES_REQUESTED** [review]

**Comment by risicle** [comment]
Bah can't test on darwin because of `Babel`'s current brokenness https://hydra.nixos.org/build/104566946

File: pkgs/tools/misc/trac/default.nix (hunk 3 above)
Comment (mmahut) [reply pkgs/tools/misc/trac/default.nix]: I think this is ```BSD License 2```, the third one mentions advertising materials or am I wrong?

https://en.wikipedia.org/wiki/BSD_licenses#3-clause_license_(%22BSD_License_2.0%22,_%22Revised_BSD_License%22,_%22New_BSD_License%22,_or_%22Modified_BSD_License%22)

IANAL, please advise if you can.

**Comment by mmahut** [comment]
> Bah can't test on darwin because of `Babel`'s current brokenness https://hydra.nixos.org/build/104566946

Indeed, will give it few minutes and try to fix it. If not, I will mark it as broken on darwin.

File: pkgs/tools/misc/trac/default.nix (hunk 3 above)
Comment (vyp) [reply pkgs/tools/misc/trac/default.nix]: Indeed, it's a bit confusing, BSD License 2 = 3-Clause BSD License = [`licenses.bsd3`](https://github.com/NixOS/nixpkgs/blob/47bd85469fbd507c12d8ec7f9650ab6dcbd8c274/lib/licenses.nix#L98) in nixpkgs.

`licenses.bsd2` is the 2-Clause BSD License, which is the next license down on your wikipedia link. Or in other words, `licenses.bsd2` != BSD License 2, which is the source of confusion. :sweat_smile:

Comment (mmahut) [reply pkgs/tools/misc/trac/default.nix]: Ooooh, got it - thank you! :)

**Comment by risicle** [comment]
I'm not sure marking as broken is necessary - it's not this package's fault and "broken" markers have a habit of hanging around longer than necessary.

**Comment by mmahut** [comment]
@risicle I agree, I just exploded it in platforms.

**Pull request merged**